lib/markbind/lib/parser.js

```diff
@@ -90,13 +90,29 @@ Parser.prototype._preprocess = function (node, context, config) {
   const self = this;
   element.attribs = element.attribs || {};
   element.attribs[ATTRIB_CWF] = path.resolve(context.cwf);
+
+  const hasSrc = _.hasIn(element.attribs, 'src');
+  if (!hasSrc) {
+    if (element.children && element.children.length > 0) {
+      element.children = element.children.map(e => self._preprocess(e, context, config));
```

Comment:
Hmm, it seems unavoidable to do this below as the last `else if`.
I don't feel like "unnecessary computation" is good enough reason to duplicate this here.

In fact, if this check is done here, it makes #174 useless.

Reply:
I duplicated this here so that nodes without a `src` attribute can exit the function early. Some of the subsequent computation e.g. `filePath` will throw an error if there is no `src` e.g. a normal `div`.

I was thinking of having a list of elements that require source (`dynamic-panel`, `include`, not sure about `panel`) before this. If those `src` are empty/null/undefined, return the error node.

Also after the file path and boilerplate is checked, return an error node if the file cannot be found. This should stop missing dynamically included files from crashing build/live reload.

Reply:
> I duplicated this here so that nodes without a `src` attribute can exit the function early. Some of the subsequent computation e.g. `filePath` will throw an error if there is no `src` e.g. a normal `div`.

Perhaps `filePath` can be conditionally computed.

> I was thinking of having a list of elements that require source (`dynamic-panel`, `include`, not sure about `panel`) before this. If those `src` are empty/null/undefined, return the error node.
>
> Also after the file path and boilerplate is checked, return an error node if the file cannot be found. This should stop missing dynamically included files from crashing build/live reload.

Sounds alright. [`panel`](https://markbind.github.io/vue-strap/#accordion) doesn't require `src`.

Reply:
Duplication still here? This is preferable:

> Perhaps `filePath` can be conditionally computed.